controlpanel/api/models/tool.py

```diff
@@ -49,17 +49,24 @@ def create(self, *args, **kwargs):
         return tool_deployment
 
     def filter(self, **kwargs):
+        deployed_versions = {}
         user = kwargs.get('user')
         filter = Q(chart_name=None)  # Always False
         for deployment in cluster.list_tool_deployments(user):
-            _, version = deployment.metadata.labels["chart"].rsplit("-", 1)
+            chart_name, version = deployment.metadata.labels["chart"].rsplit("-", 1)
+            deployed_versions[chart_name] = version
             filter = filter | (
-                Q(chart_name=deployment.metadata.labels["app"])
+                Q(chart_name=chart_name)
                 # & Q(version=version)
             )
 
         tools = Tool.objects.filter(filter)
-        return [ToolDeployment(tool, user) for tool in tools]
+        results = []
+        for tool in tools:
+            outdated = tool.version != deployed_versions[tool.chart_name]
```

Comment:
Not too comfortable with `!=`, although I experimented with a couple of semver python modules and they're all a bit unpleasant, so stick with this (at least for now)